Source/NSOrderedSet.m

```diff
@@ -0,0 +1,1912 @@
+{
+  id objs[] = {obj};
+  
+  self = [self initWithObjects: objs count: 1];
+  if(self == nil)
+    {
+      NSLog(@"Problem initializing with one element");
+    }
+  return self;
+}
+
+- (instancetype) initWithObjects:(id)firstObject, ...
+{
+  GS_USEIDLIST(firstObject,
+    self = [self initWithObjects: __objects count: __count]);
+  return self;
+}
+
+/** <init /> <override-subclass />
+ * Initialize to contain (unique elements of) objects.<br />
+ * Calls -init (which does nothing but maintain MacOS-X compatibility),
+ * and needs to be re-implemented in subclasses in order to have all
+ * other initialisers work.
+ */
+- (instancetype) initWithObjects:(const id [])objects // required override.
+                           count:(NSUInteger)count
+{
+  self = [self init];
+  if(self != nil)
+    {
+      // Need proper implementation in subclass since that is where data will be stored.
+    }
+  return self;
+}
+
+- (instancetype) initWithOrderedSet:(NSOrderedSet *)aSet
+{
+  return [self initWithOrderedSet: aSet copyItems: NO];
+}
+
+- (instancetype) initWithOrderedSet:(NSOrderedSet *)other
+                          copyItems:(BOOL)flag
+{
+  unsigned	c = [other count];
+  id		o, e = [other objectEnumerator];
+  unsigned	i = 0;
+  GS_BEGINIDBUF(os, c);
+
+  while ((o = [e nextObject]))
+    {
+      if (flag)
+	os[i] = [o copy];
+      else
+	os[i] = o;
+      i++;
+    }
+  self = [self initWithObjects: os count: c];
+  if (flag)
+    {
+      while (i--)
+        {
+          [os[i] release];
+        }
+    }
+  GS_ENDIDBUF();
+  return self;
+}
+
+- (instancetype) initWithOrderedSet:(NSOrderedSet *)other
+                              range:(NSRange)range
+                          copyItems:(BOOL)flag
+{
+  unsigned	c = [other count];
+  id		o, e = [other objectEnumerator];
+  unsigned	i = 0, j = 0;
+  unsigned      loc = range.location;
+  unsigned      len = range.length;
+  GS_BEGINIDBUF(os, c);
+
+  while ((o = [e nextObject]))
+    {
+      if(i >= loc && j < len)
+	{
+	  if (flag)
+	    os[i] = [o copy];
+	  else
+	    os[i] = o;
+	  j++;
+	}
+      i++;
+
+      if(j >= len)
+	{
+	  break;
+	}
+    }
+  
+  self = [self initWithObjects: os count: c];
+  if (flag)
+    {
+      while (i--)
+        {
+          [os[i] release];
+        }
+    }
+  GS_ENDIDBUF();
+  return self;
+}
+
+- (instancetype) initWithSet:(NSSet *)aSet
+{
+  return [self initWithSet: aSet copyItems: NO];
+}
+
+- (instancetype) initWithSet:(NSSet *)other copyItems:(BOOL)flag
+{
+    unsigned	c = [other count];
+  id		o, e = [other objectEnumerator];
+  unsigned	i = 0;
+  GS_BEGINIDBUF(os, c);
+
+  while ((o = [e nextObject]))
+    {
+      if (flag)
+	os[i] = [o copy];
+      else
+	os[i] = o;
+      i++;
+    }
+  self = [self initWithObjects: os count: c];
+  if (flag)
+    {
+      while (i--)
+        {
+          [os[i] release];
+        }
+    }
+  GS_ENDIDBUF();
+  return self;
+}
+
+- (instancetype) init
+{
+  self = [super init];
```

Comment:
I think we should use the subclass implementation here and call [self initWithObjects: NULL count: 0]

Reply:
done.

Reply:
done.